### PR TITLE
When user bans or un-bans someone, notify that it will restart the app

### PR DESCRIPTION
**Pull request by Tyuoli**
...that it will restart the app

 Related to #2437

### CODE REVIEW

File: vector/src/main/java/im/vector/util/RoomUtils.java [hunk 1]
```diff
@@ -671,6 +677,36 @@ public void onClick(DialogInterface dialog, int which) {
...
+                        Intent intent = new Intent(context, SplashActivity.class);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                        context.startActivity(intent);
+                        dialog.dismiss();
+                    }
+                })
+                .setNegativeButton(R.string.do_not_restart_app, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.dismiss();
```
Comment (ganfra) [inline vector/src/main/java/im/vector/util/RoomUtils.java]: Not necessary, it already dismiss by default

File: vector/src/main/java/im/vector/util/RoomUtils.java [hunk 2]
```diff
@@ -671,6 +677,36 @@ public void onClick(DialogInterface dialog, int which) {
...
+                .setPositiveButton(R.string.restart_app_now, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        if (null != EventStreamService.getInstance()) {
+                            EventStreamService.getInstance().stopNow();
+                        }
+                        Intent intent = new Intent(context, SplashActivity.class);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                        context.startActivity(intent);
+                        dialog.dismiss();
```
Comment (ganfra) [inline vector/src/main/java/im/vector/util/RoomUtils.java]: Not necessary, it already dismiss by default

File: vector/src/main/java/im/vector/util/RoomUtils.java [hunk 3]
```diff
@@ -671,6 +677,36 @@ public void onClick(DialogInterface dialog, int which) {
...
+     * @param context
+     */
+    public static void showBingRulesUpdatingDialog(final Context context) {
+        new AlertDialog.Builder(context)
+                .setMessage(R.string.room_settings_update_notification_rules)
+                .setPositiveButton(R.string.restart_app_now, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        if (null != EventStreamService.getInstance()) {
+                            EventStreamService.getInstance().stopNow();
```
Comment (ganfra) [inline vector/src/main/java/im/vector/util/RoomUtils.java]: Are you sure about the consequences of this ? Is it restarted then ?

File: vector/src/main/java/im/vector/util/RoomUtils.java [hunk 4]
```diff
@@ -671,6 +677,36 @@ public void onClick(DialogInterface dialog, int which) {
...
+                        }
+                        Intent intent = new Intent(context, SplashActivity.class);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (ganfra) [inline vector/src/main/java/im/vector/util/RoomUtils.java]: You should use intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK) instead, to restart the whole app.

**Comment by Tyuoli** [comment]
capture d ecran 2018-08-06 a 17 37 37

**Pull request closed without merging**